fargate.tf

```diff
@@ -0,0 +1,10 @@
+module "fargate" {
```

Comment:
Could you provide some guidance on how to best achieve that?

Reply:
[`node_groups.tf`](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/v12.0.0/node_groups.tf) in the root of the module has one example. But honestly I think the rejected work in #867 is a more "terraform native" way of implementing it. Makes the module easier to use stand-alone as well. My mistake as I didn't think depending on vars had been implemented yet when I moved managed node groups to a module.

Reply:
I've implemented this like in `node_groups.tf` for consistency.

Reply:
Creating this dependency creates a cycle. Are you sure the fargate module would cause the aws-auth ConfigMap to be created?

```
Error: Cycle: data.null_data_source.fargate, module.fargate.var.cluster_name, module.fargate.aws_iam_role.eks_fargate_pod, module.fargate.output.aws_auth_roles, local.configmap_roles, kubernetes_config_map.aws_auth
```